trillian/integration/ct_integration.go

```diff
@@ -179,25 +177,23 @@ func (t *testInfo) checkInclusionOf(ctx context.Context, chain []ct.ASN1Cert, sc
 			Extensions: sct.Extensions,
 		},
 	}
-	leafData, err := tls.Marshal(leaf)
+	leafHash, err := ct.LeafHashForLeaf(&leaf)
 	if err != nil {
-		return fmt.Errorf("tls.Marshal(leaf[%d])=(nil,%v); want (_,nil)", 0, err)
+		return fmt.Errorf("ct.LeafHashForLeaf(leaf[%d])=(nil,%v); want (_,nil)", 0, err)
 	}
-	leafHash := sha256.Sum256(append([]byte{ct.TreeLeafPrefix}, leafData...))
 	rsp, err := t.client().GetProofByHash(ctx, leafHash[:], sth.TreeSize)
 	t.stats.done(ctfe.GetProofByHashName, 200)
 	if err != nil {
 		return fmt.Errorf("got GetProofByHash(sct[%d],size=%d)=(nil,%v); want (_,nil)", 0, sth.TreeSize, err)
 	}
-	if err := Verifier.VerifyInclusionProof(rsp.LeafIndex, int64(sth.TreeSize), rsp.AuditPath, sth.SHA256RootHash[:], leafData); err != nil {
-		return fmt.Errorf("got VerifyInclusionProof(%d, %d,...)=%v", 0, sth.TreeSize, err)
+	if err := Verifier.VerifyInclusionProofByHash(rsp.LeafIndex, int64(sth.TreeSize), rsp.AuditPath, sth.SHA256RootHash[:], leafHash[:]); err != nil {
```

Comment:
Ah!  Handy that there's a 'ByHash' version!

Reply:
Yeah, I went digging -- it seemed odd that the serialized leaf would actually be needed.